src/crimson/admin/admin_socket.cc

```diff
@@ -271,13 +266,16 @@ seastar::future<> AdminSocket::stop()
     return seastar::now();
   }
   server_sock->abort_accept();
-  server_sock.reset();
   if (connected_sock) {
     connected_sock->shutdown_input();
     connected_sock->shutdown_output();
-    connected_sock.reset();
   }
-  return stop_gate.close();
+  return stop_gate.close().then([this] {
+    return task->then([] {
```

Comment:
Check whether the task has value?

Reply:
`task` is initialized before `server_sock` is set. so, presumably, it should have value by then. i added an `assert()` to verify this. could you take another look?